### PR TITLE
Add the User change.

### PR DESCRIPTION
**Pull request by ferchuhill**
add the schema and seed. Also, change the cloud function, add the functionality create, update and delete.

### CODE REVIEW

File: src/cloud/classes/RecyclingCenter.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  static async beforeSave(request: Parse.Cloud.BeforeSaveRequest): Promise<any> {
+
+    const recyclingCenter = request.object;
+
+    const roleACL = new Parse.ACL();
+    roleACL.setPublicReadAccess(true);
+    roleACL.setPublicWriteAccess(true);
+    const role: Parse.Role = new Parse.Role('ROLE_'+recyclingCenter.get("name"),roleACL);
+
+    await role.save(null,
```
Comment (santiagosemhan) [inline src/cloud/classes/RecyclingCenter.ts:18]: check format here

File: src/cloud/classes/RecyclingCenter.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        useMasterKey: true,
+      });
+
+    return request.object;
+  }
+
+  static async afterDelete(request: Parse.Cloud.AfterDeleteRequest): Promise<any> {
+    const recyclingCenter = request.object;
+    const user = recyclingCenter.get('user');
+    await UserService.clearUserSessions(user);
```
Comment (santiagosemhan) [inline src/cloud/classes/RecyclingCenter.ts:29]: ???

File: src/cloud/controllers/UserController.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { UserService } from '../services';
```
Comment (santiagosemhan) [inline src/cloud/controllers/UserController.ts:1]: Review function names

File: src/cloud/services/UserService.ts [hunk 4]
```diff
@@ -23,8 +28,197 @@ const findUserById = async (id: string): Promise<Parse.User> => {
   }
 };
 
+const findUserAdmin = async (displayLimit: number, page: number): Promise<Parse.Object[]> => {
```
Comment (santiagosemhan) [inline src/cloud/services/UserService.ts:31]: Review function names, and check if the user is admin based on the current user.

**Review by ferchuhill — COMMENTED** [review]
format the code, change the name of the services, and add the function to check if the user is admin

File: src/cloud/services/UserService.ts [hunk 5]
```diff
@@ -23,8 +27,248 @@ const findUserById = async (id: string): Promise<Parse.User> => {
...
+      emailVerified,
+      factory,
+      firstName,
+      middleName,
+      lastName,
+      nickname,
+      aboutMe,
+    } = data;
+
+    if (factory !== undefined) {
```
Comment (santiagosemhan) [inline src/cloud/services/UserService.ts:172]: replace with if (factory)

**Review by ferchuhill — COMMENTED** [review]
replace  syntax with undefined in the userService

**Comment by ferchuhill** [comment]
replace syntax with undefined in the roleService

File: src/cloud/classes/index.ts [hunk 6]
```diff
@@ -4,7 +4,8 @@ import Post from './Post';
 import Container from './Container';
 import Transaction from './Transaction';
 import TransactionDetail from './TransactionDetail';
+import RecyclingCenter from './RecyclingCenter';
 
 export {
-  Account, Address, Post, Container, Transaction, TransactionDetail,
+  Account, Address, Post, Container, Transaction, TransactionDetail,RecyclingCenter
```
Comment (santiagosemhan) [inline src/cloud/classes/index.ts:10]: Check space after the last comma.

**Review by santiagosemhan — CHANGES_REQUESTED** [review]

**Comment by ferchuhill** [comment]
change the schema the class user and role

**Review by santiagosemhan — APPROVED** [review]

**Pull request merged**